### PR TITLE
feat(clerk-js): Fallback to invisible CAPTCHA if the element is not found

### PR DESCRIPTION
**Pull request by anagstef**
## Description

This PR:

- Expects a `captcha_public_key_invisible` CAPTCHA key alongside with the main key, so it can fallback to `invisible` CAPTCHA if the `clerk-captcha` element is not found in the DOM.
- Utilizes the `unsupported-callback` prop of the CF Turnstile, to show error when the browser is not supported.
- Sends a `captcha_widget_type` prop in the `/sign_ups` POST request to FAPI to let it know what widget type was used.
- Removes the support to `always_visible` widget type option.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/types/src/displayConfig.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import type { DisplayThemeJSON } from './json';
 import type { ClerkResource } from './resource';
 
 export type PreferredSignInStrategy = 'password' | 'otp';
-export type CaptchaWidgetType = 'smart' | 'always_visible' | 'invisible' | null;
```
Comment (dimkl) [inline packages/types/src/displayConfig.ts:5]: ❓ Did we drop this option? If so, shouldn't we mention it in changelog that the `always_visible` is removed?

**Review by dimkl — APPROVED** [review]
❓ How did the `captchaPublicKeyInvisible` was populated up until now? Wasn;t the `invisible` mode the default up until now?
I am mostly asking because it seems that this is a fix, not a feat :/

File: packages/types/src/displayConfig.ts (hunk 1 above)
Comment (dimkl) [reply packages/types/src/displayConfig.ts:5]: since it wasn't introduced publicly, you are correct, we shouldn't add it in the changelog. nice!

**Comment by anagstef** [comment]
> ❓ How did the captchaPublicKeyInvisible was populated up until now? Wasn;t the invisible mode the default up until now?

@dimkl you are right, the default one until now was Invisible. This `captchaPublicKeyInvisible` new property is actually a fallback that will be used if the main captcha strategy cannot be used. I'll add a comment to explain this.

**Comment by panteliselef** [comment]
I'm confused as to why we need 2 site keys at the same time

**Comment by anagstef** [comment]
@panteliselef The `captchaPublicKeyInvisible` is a fallback to invisible, if the `clerk-captcha` element is not found in the DOM.

**Review by panteliselef — APPROVED** [review]

**Review by nikosdouvlis — APPROVED** [review]

**Pull request merged**